### ISSUE TITLE
note: the compiler unexpectedly panicked. this is a bug.

### ISSUE THREAD

**Issue by emabee** [rust-lang/rust#25877]
```rust
fn main() {
    let mut a = [0;10];
    a[5] = 5;

    let slice = & a[4..7];
    for i in 0..3 {
        println!("slice[{0}]:{1}",i,slice[i]);
    } 
    
    take(a);

    for i in 0..10 {
        println!("a[{0}]:{1}",i,a[i]);
    }

}


fn take (a [i32]){
//
}
```

**Comment by oli-obk** [comment]
I cannot reproduce the ICE on nightly or stable, please provide the error message and compiler-version. Also your example is missing a `:` between the `a` and the `[i32]` of the `take` function

**Comment by emabee** [comment]
$ rustc --version
rustc 1.0.0-beta (9854143cb 2015-04-02) (built 2015-04-02)

$ cargo run
   Compiling test v0.0.1 (file:///C:/dev/rust/projects/test)
src\main.rs:19:10: 19:11 error: internal compiler error: ident only path should have been covered already
src\main.rs:19 fn take (a [i32]){
                        ^
note: the compiler unexpectedly panicked. this is a bug.
note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports
note: run with `RUST_BACKTRACE=1` for a backtrace
thread 'rustc' panicked at 'Box<Any>', C:/bot/slave/beta-dist-rustc-win-64/build/src/libsyntax\diagnostic.rs:130


Could not compile `test`.

To learn more, run the command again with --verbose.

====
> Also your example is missing a : between the a and the [i32] of the take function
Sure. But if I add it, rustc doesn't panic anymore. I'm just running through the tutorials, and am wildly trying things out. :-)

**Comment by oli-obk** [comment]
please update your rust compiler. I believe this has been fixed in #22544

**Comment by alexcrichton** [comment]
Confirmed that this is fixed on stable and beyond, but thanks for the report!
